src/types.rs

```diff
@@ -483,22 +483,16 @@ impl Rewrite for ast::Ty {
                 ty.rewrite(context, budget, offset + 1).map(|ty_str| format!("({})", ty_str))
             }
             ast::TyTup(ref tup_ret) => {
-                let inner = try_opt!(tup_ret.iter()
-                                            .map(|item| item.rewrite(context, width, offset))
-                                            .fold(Some("".to_owned()),
-                                                  |sum, x| {
-                                                      match (sum, x) {
-                                                          (Some(sum), Some(x)) => {
-                                                              if sum == "" {
-                                                                  // First item.
-                                                                  Some(x)
-                                                              } else {
-                                                                  Some(sum + ", " + &x)
-                                                              }
-                                                          }
-                                                          _ => None,
-                                                      }
-                                                  }));
+                let inner = if let [ref item] = &**tup_ret {
+                    try_opt!(item.rewrite(context, width, offset).map(|rewrite| rewrite + ","))
+                } else {
+                    let rewrites: Option<Vec<_>>;
+                    rewrites = tup_ret.iter()
```

Comment:
Oh I didn't know that FromIterator already returns None in case 1 or more elements is None, you're right, discard what I said earlier.

On an unrelated note, how did you find that information? I don't seem to find it in the docs.